.ci/Jenkinsfile

```diff
@@ -343,6 +345,19 @@ pipeline {
             }
           }
         }
+        stage('Publish to npm') {
+          steps {
+            withNodeJSEnv(version: 'v14.17.5') {
+              withNpmrc(secret: "${env.NPMRC_SECRET}", path: "${env.WORKSPACE}/${env.BASE_DIR}") {
+                withTotpVault(secret: "${env.TOTP_SECRET}", code_var_name: 'TOTP_CODE') {
+                  dir("${BASE_DIR}") {
+                    sh(label: 'npm-publish', script: 'npm publish --otp ${TOTP_CODE}')
```

Comment:
I opened https://github.com/elastic/apm-agent-nodejs/issues/2668 for a future concern with tags/releases for maintenance releases that are not for the current major version.

Do you think it would be worthwhile to change:

```
    stage('Release') {
      options { skipDefaultCheckout() }
      when {
        beforeAgent true
        tag pattern: 'v\\d+\\.\\d+\\.\\d+', comparator: 'REGEXP'
```

to:

```
    stage('Release') {
      options { skipDefaultCheckout() }
      when {
        beforeAgent true
        tag pattern: 'v3\\.\\d+\\.\\d+', comparator: 'REGEXP'
```

to lock this "Release" stage down to new tags of the *current* major only?

If when we have a new 4.x major version and do occasional 3.x maint releases, we won't want all of these release stages to run, or at least not in the same way.

Reply:
I will work as you expect. What steps you wish to exclude for maintenance releases?

Reply:
I'd like to hear an opinion from @v1v (who initially added the 'tag pattern' usage) if restricting this to the just tags for the current major version sounds reasonable.

Reply:
I wonder if this proposal should be done only when those potential release stages for the 3.x maint releases are in place? Otherwise, there is a chance we don't enable them and then `v3` will be hardcoded in the pipeline when `v4` is branched off.

Reply:
A question about Jenkinsfile usage: If/when we have a "3.x" maintenance branch and "main" is used for 4.x releases of this agent, the only relevant ".ci/Jenkinsfile" will be the one on "main", right? If, for example, I push a "v3.42.45" tag that points to a commit on the "3.x" branch, will it still be the ".ci/Jenkinsfile" on the **"main" branch** that is used by Jenkins to configure the build?

Reply:
Thanks for clarifying my misunderstanding. That answers my concerns for this point on the `tag pattern` usage.